pages/[type]/[identifier]/edit.js

```diff
@@ -0,0 +1,210 @@
+import Request from "../../../libs/request"
+import EditJson from "../../../components/edit-func"
+import PropTypes from 'prop-types'
+import React from "react"
+import { useContext, useState } from "react"
+import { Button, LinkButton } from "../../../components/button"
+import { API_URL } from "../../../libs/constants"
+import { canEdit } from "../../../libs/general"
+import SessionContext from '../../../components/session-context'
+import { useRouter } from 'next/router'
+
+/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "^_.*" }]*/
+
+const updateItem = async (path, item, session) => {
+    const response = await fetch(`${API_URL}${path}`, {
+        method: "PATCH",
+        credentials: "include",
+        headers: {
+            "Content-Type": "application/json",
+            "X-CSRF-Token": session._csrft_,
+        },
+        body: JSON.stringify(item),
+    }).then(response => {
+        return response
+    }).catch(_err => {
+        return {
+            errors: [
+                {
+                    description: "Network error while saving",
+                    names: ["unknown"],
+                },
+            ],
+        }
+    })
+    return response
+}
+
+const SaveCancelControl = ({ cancelClick, saveClick, itemPath, saveEnabled }) => {
+    return (
+        <div className="flex space-x-1">
+            <LinkButton href={itemPath} navigationClick={cancelClick}>Cancel</LinkButton>
+            <Button onClick={saveClick} enabled={saveEnabled} type={ saveEnabled ? "primary" : "info" }>Save</Button>
+        </div>)
+}
+
+SaveCancelControl.propTypes = {
+    cancelClick: PropTypes.func.isRequired,
+    saveClick: PropTypes.func.isRequired,
+    itemPath: PropTypes.string.isRequired,
+    saveEnabled: PropTypes.bool.isRequired,
+}
+
+const filterItem = (item) => {
+    const {
+        ['@type']: _type,
+        ['@id']: _id,
+        ['@context']: _context,
+        ['uuid']: _uuid,
+        ['actions']: _actions,
+        ['access_keys']: _access,
+        ['schema_version']: _version,
+        ['title']: _title,
+        ...filtered } = item
+    return filtered
+}
```

Comment:
Oh, I missed this before. I think this is what you asked about before, where you wanted to copy an object but with certain properties filtered out. I think you should do a fetch of the object with ?frame=edit so that the back end can determine what it does and doesn’t want in a writeable object, instead of filtering an object on the front end.

Reply:
oh yes I can change that as well then